src/BugsnagUnity/Client.cs

```diff
@@ -55,6 +55,21 @@ internal class Client : IClient
 
         private bool _isUnity2019OrHigher;
 
+        private static Stopwatch _bsTimeStopwatch;
+        // A simple timer used internally to avoid thread complications when using the unity Time API
+        internal static double BSTime
+        {
+            get
+            {
+                if (_bsTimeStopwatch == null)
+                {
+                    _bsTimeStopwatch = new Stopwatch();
+                    _bsTimeStopwatch.Start();
+                }
+                return _bsTimeStopwatch.Elapsed.TotalSeconds;
+            }
+        }
```

Comment:
Does it possibly make sense to move this into its own class (especially since it's `static`)?

I'm also a bit unsure of the unprotected lazy-init. It looks like we might require some protection around `_bsTimeStopwatch` to be sure that we don't accidentally create more than one. 🤔 

This could either be a double-checked lock:

```
if (_bsTimeStopwatch == null)
{
    lock(lockObject)
    {
        if (_bsTimeStopwatch == null) {
            // init stopwatch here
```

or we could use an atomic reference, but that would require some additional overhead every time we referenced `_bsTimeStopwatch`.

Reply:
good thinking, done 👍